### PR TITLE
feat: Federated Integ Test automated fab script

### PR DESCRIPTION
**Pull request by uri200**
Signed-off-by: Oriol Batalla <obatalla@fb.com>

## Summary

This PR adds some commands to lte/fabfile to build and run `federated_integ_test`. 

We also have added an entry to show how to build and run the test on our documentation that will be seen at `Technical Reference/FEG`

Note we currently only support running this test using AGW over Vagrant VM. In the future we may change that to support non vagrant agw.

## Test Plan
Run test locally
```
cd magma/lte/gateway
fab federated_integ_test
```

Checked new manual on docusaurus docker container
image

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/python/integ_tests/federated_tests/s1aptests/test_attach_detach.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
```
Comment (uri200) [inline lte/gateway/python/integ_tests/federated_tests/s1aptests/test_attach_detach.py:12]: This script is a copy of the `test_attach_detach.py` from lte integ test.

We will modify it in the future once we configure HSS remotely (right now it is configured manually)

File: docs/readmes/feg/s1ap_federated_test.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+- Start AGW, Orc8r and FEG
+- Configure Orc8r with AGW and FEG
+- Run connectivity tests between all 3 components
+- Run a basic attach/detach test
+
+```
+cd magma/lte/gatewat
+fab federated_integ_test
+```
+
```
Comment (uri200) [inline docs/readmes/feg/s1ap_federated_test.md:61]: after this line, the text below was previously on the README. I just moved it here

File: lte/gateway/fabfile.py [hunk 3]
```diff
@@ -217,6 +219,31 @@ def s1ap_setup_cloud():
...
+    vagrant_setup(
+        'magma_trfserver', destroy_vm, force_provision=provision_vm)
+
+    vagrant_setup(
+            'magma_test', destroy_vm, force_provision=provision_vm,
+        )
+    execute(_make_integ_tests)
+    execute(run_integ_tests, "federated_tests/s1aptests/test_attach_detach.py")
+
+
```
Comment (uri200) [inline lte/gateway/fabfile.py:250]: this is just the steps to run the test. Note they are different than integ_test because they only support Vagrant AGW for now

**Review by koolzz — APPROVED** [review]

File: lte/gateway/fabfile.py [hunk 4]
```diff
@@ -310,7 +337,8 @@ def run_integ_tests(tests=None):
     command in test machine:
     $ make integ_test TESTS=s1aptests/test_attach_detach.py
     """
-    test_host = vagrant_setup("magma_test", destroy_vm=False)
+    test_host = vagrant_setup("magma_test", destroy_vm=False,
```
Comment (ardzoht) [inline lte/gateway/fabfile.py]: what is the reason for this change?

Comment (uri200) [reply lte/gateway/fabfile.py]: yes and not: it looks like vagrant_setup was defaulting `force_provision` to string `False`., which caused to be True on the comparison

I am going to change instead force_provision at vagrant_setup to use boolean value by default. 

This is the problem with types on fab: the arguments we pass through the command line always come as strings.

**Review by ardzoht — APPROVED** [review]
lgtm

**Review by emakeev — APPROVED** [review]

**Review by electronjoe — APPROVED** [review]

**Review by themarwhal — APPROVED** [review]
